cmd/logging/logging.go

```diff
@@ -0,0 +1,78 @@
+package logging
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"github.com/Microsoft/go-winio"
+	"io"
+	"os"
+	"os/signal"
+)
+
+type Config struct {
+	ID string
+	Namespace string
+	Stdout io.Reader
+	Stderr io.Reader
+}
+
+type LoggerFunc func(context.Context, *Config, func() error) error
+
+func Run(fn LoggerFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	var (
+		s	= make(chan os.Signal, 31)
+		errCh	= make(chan error, 0)
+	)
+
+	var serr io.Reader
+	sout, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_STDOUT"))
+	// serr, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_STDERR"))
+	wait, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_WAIT"))
+
+	config := &Config{
+		ID:	os.Getenv("CONTAINER_ID"),
+		Namespace: os.Getenv("CONTAINER_NAMESPACE"),
+		Stdout: sout,
+		Stderr: serr,
+	}
+
+	// TODO: Need to double check what the actual signal should be here
+	signal.Notify(s)
+	f, _ := os.Create("C:\\Users\\Administrator\\LCOW\\binary-results.txt")
+	w := bufio.NewWriter(f)
+	defer f.Close()
+
+	go func() {
+		if err := fn(ctx, config, wait.Close); err != nil {
+			w.WriteString("Binary exited with error. sending error via channel\n")
+			w.Flush()
+			errCh <- err
+			return
+		}
+		w.WriteString("Binary exited normally.\n")
+		w.Flush()
+		errCh <- nil
+	}()
```

Comment:
The `logger` function takes in wait.Close as the parameter `ready` and this is called at the end of that function. Why then does this function here call the logger function and additionally later call `Close` on wait again?

Reply:
good point. replied above. changed the implementation a bit